app/api/probe_dock/reports_api.rb

```diff
@@ -116,6 +122,31 @@ def serialization_options reports
             paginated_rel = paginated_rel.where('categories.name in (?)', params[:categoryNames].collect(&:to_s).to_a)
           end
 
+          if array_param?(:status) || array_param?(:kind)
```

Comment:
This should be split into two `if` blocks. It currently is bugged if you combine status and kind parameters.

If you search for `passed` status and `new` kind, you might get payloads that do not contain any passed tests but that do contain new tests. This is inconsistent with the overall API behavior where separate parameters are always "AND-ed".